app/screens/edit_post/edit_post.tsx

```diff
@@ -0,0 +1,241 @@
+        const unsubscribe = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
+                switch (buttonId) {
+                    case LEFT_BUTTON.id: {
+                        onClose();
+                        break;
+                    }
+                    case RIGHT_BUTTON.id:
+                        onSaveEditedPost();
+                        break;
+                }
+            },
+        }, componentId);
+
+        return () => {
+            unsubscribe.remove();
+        };
+    }, [postMessage]);
+
+    useDidUpdate(() => {
+        // Workaround to avoid iOS emdash autocorrect in Code Blocks
+        if (Platform.OS === 'ios') {
+            onPostSelectionChange();
+        }
+    }, [postMessage]);
+
+    const onPostSelectionChange = useCallback((curPos: number = cursorPosition) => {
+        if (Platform.OS === 'ios') {
+            setKeyboardType(switchKeyboardForCodeBlocks(postMessage, curPos));
+        }
+        setCursorPosition(curPos);
+    }, [cursorPosition, postMessage]);
+
+    const updateCanEditPostButton = useCallback((enabled) => {
+        if (rightButtonEnabled !== enabled) {
+            setRightButtonEnabled(enabled);
+            setButtons(componentId, {
+                leftButtons: [{...LEFT_BUTTON, icon: closeButtonIcon}],
+                rightButtons: [{...RIGHT_BUTTON, enabled}],
+            });
+        }
+    }, [closeButtonIcon, rightButtonEnabled, componentId]);
+
+    const onPostChangeText = useCallback((message: string) => {
+        setPostMessage(message);
+        const tooLong = message.trim().length > maxPostSize;
+
+        if (tooLong) {
+            const line = tooLong ? intl.formatMessage({id: 'mobile.message_length.message_split_left', defaultMessage: 'Message exceeds the character limit'}) : undefined;
+            const extra = tooLong ? `${message.trim().length} / ${maxPostSize}` : undefined;
+            setErrorLine(line);
+            setErrorExtra(extra);
+        }
+        updateCanEditPostButton(Boolean(message) && !tooLong);
+    }, [intl, maxPostSize, updateCanEditPostButton]);
+
+    const emitEditing = useCallback((loading) => {
+        setRightButtonEnabled(!loading);
+        setButtons(componentId, {
+            leftButtons: [{...LEFT_BUTTON, icon: closeButtonIcon}],
+            rightButtons: [{...RIGHT_BUTTON, enabled: !loading}],
+        });
+    }, [closeButtonIcon]);
+
+    const onClose = useCallback(() => {
+        Keyboard.dismiss();
+        popTopScreen();
+    }, []);
+
+    const onSaveEditedPost = useCallback(async () => {
+        setIsUpdating(true);
+        setErrorLine(undefined);
+        setErrorExtra(undefined);
+
+        emitEditing(true);
+
+        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+        // @ts-ignore
+        const {error: {status_code, message: errorMessage}} = await editPost(serverUrl, post.id, postMessage);
```

Comment:
Why are we showing the error message from the server? perhaps we need UX input here? I know in v1 we used to do that but with the server looking into removing all the localizations perhaps is best if we show a message generated in the client regardless of what the error from the server is?

Reply:
@matthewbirtch  Can you please guide us here?  If the post fails to edit, what should we display?

Reply:
>  I know in v1 we used to do that but with the server looking into removing all the localizations perhaps is best if we show a message generated in the client regardless of what the error from the server is?

I don't quite understand the issue described here. Can you elaborate or clarify why we wouldn't do what we do in v1?

One potential solution could be to use a toast in this scenario, but I need to understand the problem better

Reply:
@matthewbirtch in v1 when editing a post fails for some reason we display to the user an error message at the top of the screen.

More importantly, the error message is sent by the server and it can be either easy to understand or a bit cryptic. So the ask is a for a nice message to be displayed, regardless of where in the screen is shown. I could even go with an Alert, the way we show it is not as important as the message itself.

Reply:
Ok, got it. So you need something all encompassing? Like "There was a problem editing this message. Please try again"?

Reply:
Maybe @cwarnermm can help with the message here

Reply:
Thanks, @matthewbirtch! I like this simple approach. @enahum - is Matt's suggestion good from a screen real estate perspective? If you need something shorter, perhaps: "Something went wrong. Please try again."

Reply:
Thanks @enahum , @matthewbirtch  and @cwarnermm .  I have updated the text to the one that @matthewbirtch  suggested.